app/services/variant_units/unit_prices.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module VariantUnits
+  class UnitPrices
+    def initialize(variant)
+      @variant = variant
+      @product = variant.product
+    end
+
+    def unit_price_denominator
+      return @variant.unit_value if @product&.variant_unit == "items"
+
+      case unit_price_unit
+      when "lb"
+        @variant.unit_value / 453.6
+      when "kg"
+        @variant.unit_value / 1000
+      else # Liters
+        @variant.unit_value
+      end
```

Comment:
There's a missing case here, right?

Reply:
I think it's covered by the guard clause, but it's not very obvious. If we make it through the guard clause, it means that `product.variant_unit != "items"`, which means that `unit_price_unit` will be one of `["lb", "kg", "L"]`.

Reply:
I added a comment to that effect... but comments are 👃. If you think it needs a refactoring let me know. If I do an `when "L"` and then an `else` though, the return values are the same.

Reply:
Ah, I see. All the "items" cases are caught by the guard clause. :ok_hand: